finance/table_finance_us_sec_public_company.go

```diff
@@ -19,18 +21,87 @@ func tableFinanceUsSecPublicCompany(ctx context.Context) *plugin.Table {
 		},
 	}
 }
 
+type Company struct {
+	CIK                 string `json:"cik"`
+	Currency            string `json:"currency"`
+	Date                string `json:"date"`
+	Exchange            string `json:"exchange"`
+	ExchangeName        string `json:"exchangeName"`
+	ExchangeSegment     string `json:"exchangeSegment"`
+	ExchangeSegmentName string `json:"exchangeSegmentName"`
+	ExchangeSuffix      string `json:"exchangeSuffix"`
+	FIGI                string `json:"figi"`
+	IEXID               string `json:"iexId"`
+	IsEnabled           bool   `json:"isEnabled"`
+	LEI                 string `json:"lei"`
+	Name                string `json:"name"`
+	Region              string `json:"region"`
+	Symbol              string `json:"symbol"`
+	Type                string `json:"type"`
+	// Define other fields based on the API response
+}
+
 func listUsSecPublicCompany(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
-	companies, err := core.GetPublicCompaniesWithHeaders(map[string]string{"User-Agent": "Steampipe/v0.x"})
+
+	client := resty.New()
+
+	// We were using the SDK from github.com/piquette/edgr/core to retrieve Security Symbols Reference Data.
+	// However, the SDK is currently encountering a Forbidden error when accessing the endpoint https://api.iextrading.com/1.0/ref-data/symbols,
+	// despite using a valid API token.
+	// We raised a support request (https://github.com/piquette/edgr/issues/3) to address the error; however, it has not yet been resolved.
+	// Fortunately, the new API endpoint, as described in the documentation (https://iexcloud.io/docs/core/REF_DATA#security-symbols-reference-data),
+	// is now providing the expected data.
+	// For further discussions regarding the legacy and new API endpoints, please refer to this discussion thread: https://infinitekind.tenderapp.com/discussions/problems/57171-iex-trading.
+	apiUrl := "https://api.iex.cloud/v1/data/core/REF_DATA/symbols"
+
+	config := GetConfig(d.Connection)
+	if config.IEXAPIToken == nil {
+		return nil, fmt.Errorf("partial credentials found in connection config, missing: iex_api_token")
+	}
+
+	var pageCompanies []Company
+
+	// Making a GET request with pagination query parameters
+	resp, err := client.R().
+		SetHeaders(map[string]string{
+			"User-Agent": "Steampipe/v0.x",
+		}).
+		SetQueryParams(map[string]string{
+			"token": *config.IEXAPIToken,
+		}).
+		SetResult(&pageCompanies).
+		Get(apiUrl)
+
 	if err != nil {
-		plugin.Logger(ctx).Error("finance_us_sec_public_company.listUsSecPublicCompany", "query_error", err)
+		plugin.Logger(ctx).Error("finance_us_sec_public_company.listUsSecPublicCompany", "Error on request: ", err)
```

Comment:
```suggestion
		plugin.Logger(ctx).Error("finance_us_sec_public_company.listUsSecPublicCompany", "query_error", err)
```

The second arg should be a snake cased error type